src/couch_replicator_manager.erl

```diff
@@ -403,13 +442,10 @@ process_update(State, DbName, {Change}) ->
         end
     end.
 
-owner(<<"shards/", _/binary>> = DbName, DocId) ->
-    Live = [node()|nodes()],
+owner(<<"shards/", _/binary>> = DbName, DocId, Live) ->
     Nodes = lists:sort([N || #shard{node=N} <- mem3:shards(mem3:dbname(DbName), DocId),
 			     lists:member(N, Live)]),
-    node() =:= hd(mem3_util:rotate_list({DbName, DocId}, Nodes));
-owner(_DbName, _DocId) ->
```

Comment:
@rnewson @davisp I'm a little concerned by the fact that the second function clause for owner/2 disappears - it doesn't exist over in cloudant/couch_replicator and I wasn't quite sure what the right thing to do here was, as I don't have the context of why the second clause existed here.